frontend/lib/verification/verification_qr_content_parser.dart

```diff
@@ -0,0 +1,125 @@
+                "werden. Vermutlich handelt es sich um einen QR Code, "
+                "der nicht für die Ehrenamtskarte App generiert wurde.");
+      }
+
+      final fullName = cardVerifyModel.fullName;
+      if (fullName == null || fullName.isEmpty) {
+        _verificationCardDetailsModel.setVerificationFailure(
+            createFieldMissingError("fullName", "fullNameMissing"));
+        return QRCodeParseResult();
+      }
+      final randomBytes = cardVerifyModel.randomBytes;
+      if (randomBytes == null) {
+        _verificationCardDetailsModel.setVerificationFailure(
+            createFieldMissingError("randomBytes", "randomBytesMissing"));
+        return QRCodeParseResult();
+      }
+      final unixInt64ExpirationDate = cardVerifyModel.expirationDate;
+      int unixExpirationDate;
+      if (unixInt64ExpirationDate != null) {
+        try {
+          unixExpirationDate = unixInt64ExpirationDate.toInt();
+          if (unixExpirationDate == null) {
+            throw Exception(["unixExpirationDate was null"]);
+          }
+        } on Exception catch (e) {
+          return QRCodeParseResult(
+              hasError: true,
+              internalErrorMessage: "Failed to parse unixExpirationDate, "
+                  "Message: ${e.toString()}",
+              userErrorMessage: "Beim Verarbeiten des Ablaufdatums ist ein "
+                  "unerwarteter Fehler aufgetreten.");
+        }
+      }
+      if (cardVerifyModel.cardType == CardVerifyModel_CardType.STANDARD &&
+          unixExpirationDate == null) {
+        _verificationCardDetailsModel
+            .setVerificationFailure(VerificationError.fromStrings(
+                "Die eingescannte Karte enthält kein Ablauf-"
+                    "datum, obwohl dies für die blaue Ehrenamtskarte"
+                    " erforderlich ist.",
+                "ExpirationDateWithStandardMissing"));
+
+        return QRCodeParseResult();
+      }
+      final cardType = cardVerifyModel.cardType.toString();
+      final regionId = cardVerifyModel.region;
+      final otp = cardVerifyModel.otp;
+      if (otp == null) {
+        _verificationCardDetailsModel.setVerificationFailure(
+            VerificationError.fromStrings(
+                "Eine zur Verifikation erforderliche Information fehlt.",
+                "otpMissing"));
+        return QRCodeParseResult();
+      }
+
+      final baseCardDetails = BaseCardDetails(
+          fullName, randomBytes, unixExpirationDate, cardType, regionId);
+      final verificationCardDetails =
+          VerificationCardDetails(baseCardDetails, otp);
+      _verificationCardDetailsModel
+          .setWaitingForRemoteVerification(verificationCardDetails);
+    } on Exception catch (e) {
+      return QRCodeParseResult(
+          hasError: true,
+          internalErrorMessage: e.toString(),
+          userErrorMessage: "Ein unerwarteter Fehler ist aufgetreten.");
+    }
+    return QRCodeParseResult(hasError: false);
```

Comment:
… aaaand again, would rather use exceptions here. I have no idea what the "happy flow" does 😄